build/openresty/wasmx/rules.bzl

```diff
@@ -0,0 +1,88 @@
+load("//build/openresty/wasmx:wasmx_repositories.bzl", "wasm_runtimes")
+
+wasmx_configure_options = select({
+    "@platforms//os:linux": [
+        # neded for wasmx module
+        # although this is centos7 specific, the flag will work on any GNU linker
+        # we place it here to skip macos, which uses darwin ld
+        # https://github.com/Kong/ngx_wasm_module/commit/e70a19f53e1dda99d016c5cfa393652720959afd
+        "--with-ld-opt=\"-Wl,--allow-multiple-definition\"",
+    ],
+    "//conditions:default": [],
+}) + select({
+    "@kong//:wasmx_flag": [
+        "--with-cc-opt=\"-DNGX_WASM_HOST_PROPERTY_NAMESPACE=kong\"",
+        "--with-cc-opt=\"-I$$INSTALLDIR/../wasm_runtime/include\"",
+        "--with-ld-opt=\"-L$$INSTALLDIR/../wasm_runtime/lib\"",
+    ],
+    "//conditions:default": [],
+}) + select({
+    "@kong//:wasmx_static_mod": [
+        "--add-module=$$EXT_BUILD_ROOT$$/external/ngx_wasm_module",
+    ],
+    "@kong//:wasmx_dynamic_mod": [
+        "--with-compat",
+        "--add-dynamic-module=$$EXT_BUILD_ROOT$$/external/ngx_wasm_module",
+    ],
+    "//conditions:default": [],
+})
+
+wasmx_env = select({
+    "@kong//build/openresty/wasmx:use_v8": {
+        "NGX_WASM_RUNTIME": "v8",
+        # https://github.com/Kong/ngx_wasm_module/blob/0f07c712c48d410190ec5e0cc0b34fdfd190387d/t/10-build/003-dynamic_module.t#L43
+        "NGX_WASM_RUNTIME_LD_OPT": "$$INSTALLDIR/../wasm_runtime/lib/libwee8.a -lv8bridge -lstdc++ -lm -ldl -lpthread",
```

Comment:
All other libraries (`-lv8bridge -lstdc++ -lm -ldl -lpthread` etc...) are already automatically added when statically linking any of the 3 runtimes, they can be omitted from all `NGX_WASM_RUNTIME_LD_OPT` options. Additionally, `--with-ld-opt=\"-L$$INSTALLDIR/../wasm_runtime/lib\"", is redundant with these `NGX_WASM_RUNTIME_LD_OPT` env variables. I'd advise to only use `--with-cc/ld-opt` and remove all `NGX_WASM_RUNTIME_*` env variable, except the runtime name (`NGX_WASM_RUNTIME={v8,wasmer,wasmtime}`.

Reply:
I did a test and it seems removing `NGX_WASM_RUNTIME_LD_OPT` is ok, but removing `NGX_WASM_RUNTIME_{LIB,INC}` makes configure script to error out when checking the runtime. So i'm going to remove `NGX_WASM_RUNTIME_LD_OPT` and --with-cc/ld-opt and keep `NGX_WASM_RUNTIME_{LIB,INC}`.